hudi-common/src/main/java/org/apache/hudi/common/table/timeline/TimelineUtils.java

```diff
@@ -98,4 +101,31 @@
 
     }).distinct().filter(s -> !s.isEmpty()).collect(Collectors.toList());
   }
+
+  /**
+   * Get extra metadata for specified key from latest commit/deltacommit instant.
+   */
+  public static Option<String> getExtraMetadataFromLatest(HoodieTableMetaClient metaClient, String extraMetadataKey) {
+    return metaClient.getCommitsTimeline().filterCompletedInstants().getReverseOrderedInstants().findFirst().map(instant ->
+        getMetadataValue(metaClient, extraMetadataKey, instant)).orElse(Option.empty());
+  }
+
+  /**
+   * Get extra metadata for specified key from all active commit/deltacommit instants.
+   */
+  public static Map<String, Option<String>> getExtraMetadataTimeline(HoodieTableMetaClient metaClient, String extraMetadataKey) {
```

Comment:
@satishkotha can you describe what this method is used for ? The naming is a little confusing..

Reply:
This is primarily for debugging. This helps with seeing how extra metadata value changed over time.  If you have better suggestion for a name, let me know

Reply:
getAllExtraMetadataFromActiveTimeline ?

Reply:
@n3nash renamed. Please review